### ISSUE TITLE
Enzyme unit testing issues when wrapping an existing React component with DragDropContext

### ISSUE THREAD

**Issue by Larkenx** [react-dnd/react-dnd#1044]
I've recently implemented drag-and-drop functionality in my application by wrapping existing components. For example, one of the root components is `Session`, and I am exporting the component with the `DragDropContext`:

```javascript
export default DragDropContext(HTML5Backend)(Session)
```

However, this broke an Enzyme unit test we had in place to make sure a child React component was rendered under the `Session`, because of the `DragDropContext` wrapper.
```javascript
let wrapper = shallow(<Session ... />)
// .... 
expect(wrapper.contains(<NoteColumn ... />)).toEqual(true)
```

Do you know what's happening internally that would cause this issue? I am guessing that enzyme's `shallow` render renders a shallow `DragDropContext` react component with the `Session` component, and by doing so, does not render `Session`'s child components

**Comment by darthtrevino** [comment]
So the DragDropContext component decorates your target component. The react vdom will look like `DragDropContext` -> `Session` -> `SessionChildren`. I hope that helps!

**Comment by Larkenx** [comment]
So, I found that wrapping an existing component with `DragDropContext` broke my unit tests for a number of reasons. We are unit testing the methods on the `Session` react component. We import the component and create a new instance of that class, like so:
```javascript
let TestSession = new Session(props)
```
However, because it's exported with `DragDropContext`, `TestSession` is an instance of `DragDropContext` and does not have `Session`'s methods! This broke all of our unit tests. In the future, I suggest only wrapping your existing react components with `DragDropContext` in your higher level / parent (container) components. I moved the DragDropContext up to our `App` component, like this...
```javascript
DragDropContext(HTML5Backend)(new Session(props)).render()
```
